### PR TITLE
CY-2571-Use-Azure-SDK

### PR DESCRIPTION
**Pull request by ahmadiesa-abu**
Remove all of the old implementation of using requests library, and use the azure_sdk, some enhancements , added unit-tests

### CODE REVIEW

File: cloudify_azure/resources/app_service/plan.py [hunk 1]
```diff
@@ -12,78 +12,67 @@
...
+        ctx.logger.info("Resource with name {0} doesn't exist".format(name))
+        return
+    try:
+        plan.delete(resource_group, name)
+        utils.runtime_properties_cleanup(ctx)
+    except CloudError as cr:
+        raise cfy_exc.NonRecoverableError(
+            "delete plan '{0}' "
+            "failed with this error : {1}".format(name,
+                                                  cr.message)
```
[the remark this answers is not in the document]
Comment (ahmadiesa-abu) [reply cloudify_azure/resources/app_service/plan.py:77]: we checked the implementation of CloudError , and it override the base class , and this message will be available even on python3 https://github.com/Azure/msrestazure-for-python/blob/master/msrestazure/azure_exceptions.py#L165 , left a comment on the changelog

**Comment by ahmadiesa-abu** [comment]
@0lvin Thank you :) , changes are done

File: plugin.yaml [hunk 2]
```diff
@@ -660,7 +660,7 @@ node_types:
         type: string
         description: >
           Network API Version
-        default: "2016-03-30"
+        default: "2017-03-01"
```
[the remark this answers is not in the document]
Comment (ahmadiesa-abu) [reply plugin.yaml:663]: well , not really I went with the default one when testing , but what I can tell you that in the SDK no `2016-03-30` https://docs.microsoft.com/en-us/python/api/azure-mgmt-network/azure.mgmt.network?view=azure-python

File: cloudify_azure/utils.py [hunk 3]
```diff
@@ -18,31 +18,21 @@
...
 
 
 def dict_update(orig, updates):
     """Recursively merges two objects"""
-    for key, val in updates.items():
-        if isinstance(val, Mapping):
-            orig[key] = dict_update(orig.get(key, {}), val)
-        else:
-            orig[key] = updates[key]
+    if updates and type(updates) == dict:
```
[the remark this answers is not in the document]
Comment (ahmadiesa-abu) [reply cloudify_azure/utils.py:30]: good point :)

File: plugin.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ahmadiesa-abu) [reply plugin.yaml:663]: @EarthmanT so , I can go with the closest one to it `2016-09-01`, what do you think ?

File: constraints.txt [hunk 4]
```diff
@@ -1,4 +1,12 @@
...
+azure-mgmt-web==0.46.0
+azure-mgmt-compute==12.0.0
+azure-mgmt-containerservice==9.0.1
+azure-mgmt-network==10.1.0
+azure-mgmt-storage==9.0.0
+azure-storage-common==2.1.0
+azure-mgmt-resource==9.0.0
+azure-common==1.1.25
+msrestazure==0.6.3
+cloudify-common===4.5
```
Comment (EarthmanT) [inline constraints.txt:12]: Let's remove cloudify-common from constraints and let the setup.py do the work.

File: cloudify_azure/decorators.py [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+                    result = resource.get(resource_group_name, nsg_name, name)
+                else:
+                    result = resource.get(resource_group_name, name)
+                if result:  # found a resource with same name
+                    name = ""
+                    continue
+            except CloudError:  # this means name is not used
+                return name
+    else:
+        return name
```
[the remark this answers is not in the document]
Comment (ahmadiesa-abu) [reply cloudify_azure/decorators.py:79]: the logic has been there before [ it tries for 15 times , because uuid4 might cause collision ] so I kept it

**Pull request merged**